File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/resource/ShyvanaDragonFuryBar.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.hud.colors.BackgroundBreathingColor;
import com.bonepl.chromaleague.hud.parts.ProgressBar;
import com.bonepl.chromaleague.state.GameStateHelper;
import com.bonepl.razersdk.animation.AnimatedFrame;
import com.bonepl.razersdk.animation.Frame;
import com.bonepl.razersdk.color.BreathingColor;
import com.bonepl.razersdk.color.Color;
import com.bonepl.razersdk.color.StaticColor;

public class ShyvanaDragonFuryBar extends AnimatedFrame {
    public static final StaticColor DRAGON_FURY_COLOR = StaticColor.ORANGE;
    private static final BreathingColor ULTIMATE_READY_COLOR = new BackgroundBreathingColor(StaticColor.RED);

    @Override
    public Frame getFrame() {
        final int resourcePercentage = GameStateHelper.getResourcePercentage();
        Color color;
        if (resourcePercentage == 100) {
            color = ULTIMATE_READY_COLOR.getColor();
        } else {
            color = DRAGON_FURY_COLOR;
        }
        addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(), resourcePercentage, color));
        return super.getFrame();
    }
}

File: src/hud/parts/resource/ShyvanaDragonFuryBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {Color} from '../../../razer-sdk/color/Color.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {BackgroundBreathingColor} from '../../colors/BackgroundBreathingColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export const DRAGON_FURY_COLOR = StaticColor.ORANGE;
const ULTIMATE_READY_COLOR = new BackgroundBreathingColor(StaticColor.RED);

export class ShyvanaDragonFuryBar extends AnimatedFrame {
    pollFrame(): Frame {
        const resourcePercentage = getResourcePercentage();
        let color: Color;
        if (resourcePercentage === 100) {
            color = ULTIMATE_READY_COLOR;
        } else {
            color = DRAGON_FURY_COLOR;
        }
        this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), resourcePercentage, color));
        return super.pollFrame();
    }
}
